src/main/java/mekanism/common/tile/component/SideConfig.java

```diff
@@ -0,0 +1,56 @@
+package mekanism.common.tile.component;
+
+import net.minecraft.util.EnumFacing;
+
+import java.util.Observable;
+import java.util.Observer;
+
+public class SideConfig {
+    private byte[] directions = new byte[EnumFacing.VALUES.length];
+    private Observable observable = new Observable();
```

Comment:
Well now that I think about it there'd be no way to implement the observer list in an interface, so you either have to have a superclass or an internal object with proxy methods (which is rather annoying) 

I'll look into redoing it in a more sensible way with generics.